pkg/client/linstor.go

```diff
@@ -1134,18 +1142,23 @@ func (s *Linstor) Unmount(target string) error {
 		"target": target,
 	}).Info("unmounting volume")
 
-	notMounted, err := s.mounter.IsNotMountPoint(target)
+	notMounted, err := s.IsNotMountPoint(target)
 	if err != nil {
 		return fmt.Errorf("unable to determine mount status of %s %v", target, err)
 	}
 
-	if notMounted {
-		return nil
+	if !notMounted {
+		// Still needs to be unmounted
+		err := s.mounter.Unmount(target)
+		if err != nil {
+			return fmt.Errorf("unable to unmount target '%s': %w", target, err)
+		}
 	}
 
-	err = s.mounter.Unmount(target)
+	// We create a file/directory in Mount() which must be removed again
+	out, err := s.mounter.Run("rm", "-fd", target)
```

Comment:
we are all GNU/Linux specific anyways, here and in LINSTOR and what not, but I'd prefer a "Go solution" if it is possible. Guess that should not be too cumbersome in this case.

Reply:
I wasn't sure if that would always work: there is some magic going on in the `mounter.Run()` method regarding namespaces. But that does not seem to be active in our containers, so a normal `os.Remove()` works :shrug: